app/services/invoices/update_service.rb

```diff
@@ -35,6 +35,10 @@ def call
         return result.not_allowed_failure!(code: 'payment_status_update_on_draft_invoice')
       end
 
+      if old_payment_status.to_s == 'succeeded' && params.key?(:payment_status) && params[:payment_status].to_s != 'succeeded'
```

Comment:
I checked the file and I think we could improve a bit the logic of payment status update:

```ruby
old_payment_status = invoice.payment_status
if params.key?(:payment_status) && old_payment_status.to_s != params.key?(:payment_status).to_s
  return result.not_allowed_failure!(code: 'payment_status_update_on_draft_invoice') if invoice.draft?
  return result.not_allowed_failure!(code: 'payment_status_update_on_succeeded_invoice') if invoice.payment_succeeded?

  invoice.payment_status = params[:payment_status]
end
```